src/ring_theory/polynomial/vieta.lean

```diff
@@ -83,32 +87,43 @@ begin
 end
 
 lemma prod_X_sub_C_eq_sum_esymm (s : multiset R) :
-  (s.map (λ t, polynomial.X - polynomial.C t)).prod =
-  ∑ j in finset.range (s.card + 1),
-  (-1)^j* (polynomial.C (s.esymm j) * polynomial.X ^ (s.card - j)) :=
+  (s.map (λ t, X - C t)).prod =
+  ∑ j in finset.range (s.card + 1), (-1) ^ j * (C (s.esymm j) * X ^ (s.card - j)) :=
 begin
-  conv_lhs { congr, congr, funext, rw sub_eq_add_neg, rw ←map_neg polynomial.C _, },
+  conv_lhs { congr, congr, funext, rw sub_eq_add_neg, rw ←map_neg C _, },
   convert prod_X_add_C_eq_sum_esymm (map (λ t, -t) s) using 1,
   { rwa map_map, },
   { simp only [esymm_neg, card_map, mul_assoc, map_mul, map_pow, map_neg, map_one], },
 end
 
-lemma prod_X_sub_C_coeff (s : multiset R) {k : ℕ} (h : k ≤ s.card):
-  polynomial.coeff (s.map (λ t, polynomial.X - polynomial.C t)).prod k =
-  (-1)^(s.card - k) * s.esymm (s.card - k) :=
+lemma prod_X_sub_C_coeff (s : multiset R) {k : ℕ} (h : k ≤ s.card) :
+  (s.map (λ t, X - C t)).prod.coeff k = (-1) ^ (s.card - k) * s.esymm (s.card - k) :=
 begin
-  conv_lhs { congr, congr, congr, funext, rw sub_eq_add_neg, rw ←map_neg polynomial.C _, },
+  conv_lhs { congr, congr, congr, funext, rw sub_eq_add_neg, rw ←map_neg C _, },
   convert prod_X_add_C_coeff (map (λ t, -t) s) _ using 1,
   { rwa map_map, },
   { rwa [esymm_neg, card_map] },
   { rwa card_map },
 end
 
+/-- Vieta's formula for the coefficients and the roots of a split polynomial.
+  When `R` is a field, use `polynomial.splits_iff_card_roots` to derive the
+  condition `hroots` from `p.splits (ring_hom.id R)`.  -/
+theorem _root_.polynomial.vieta [is_domain R] (p : R[X])
```

Comment:
`p` and `k` can be implicit here, unless you have a reason for not doing it.

Also, can you add a version that uses `polynomial.splits`?

Reply:
Thanks! Please check if the new changes look good.